core/variant/array.cpp

```diff
@@ -371,25 +372,24 @@ void Array::sort() {
 }
 
 struct _ArrayVariantSortCustom {
-	Object *obj = nullptr;
-	StringName func;
+	Callable func;
 
 	_FORCE_INLINE_ bool operator()(const Variant &p_l, const Variant &p_r) const {
 		const Variant *args[2] = { &p_l, &p_r };
 		Callable::CallError err;
-		bool res = obj->call(func, args, 2, err);
+		Variant res;
+		func.call(args, 2, res, err);
 		if (err.error != Callable::CallError::CALL_OK) {
-			res = false;
+			ERR_FAIL_V_MSG(false, "Error calling sorting method: " + Variant::get_callable_error_text(func, args, 1, err));
 		}
 		return res;
 	}
 };
-void Array::sort_custom(Object *p_obj, const StringName &p_function) {
-	ERR_FAIL_NULL(p_obj);
+void Array::sort_custom(Callable p_callable) {
+	ERR_FAIL_COND(p_callable.is_null());
```

Comment:
Is users can pass `Callable()` from script, we might want to add an actual error message here to let them know that it's invalid in nicer words than the default message.

Reply:
This check might be unnecessary actually, because it's also checked in `call()` and it only fails if both method and object are null.